### DISCUSSION TITLE
I need to fire the function when keypress on cell

### DISCUSSION THREAD

**Discussion by chairedevil** [nick-keller/react-datasheet-grid#70] [Q&A] [answered]
`{ ...keyColumn('test1', textColumn), title: 'TEST ENTER KEY', component: TestEnter /* width: '0 0 300px'*/ },`

```
  const TestEnter = (data: any) => {
    if (data.focus && data.rowData.type === 0) {
      console.log('RUN function')
      data.stopEditing({ nextRow: false })
      setIsOpenDrawer(true)
    }
    return <div>{data.rowData.test2}</div>
  }
```

In my project. In the same column, some rows can call a drawer when enter key is pressed.
I used the above method to open the drawer when the enter key is pressed on the cell.
I got an error as below. Do you have a better way to accomplish this?

![スクリーンショット 2021-10-20 152106](https://user-images.githubusercontent.com/22753035/138038843-7acc1d3f-857b-4683-a126-04322e3b09e8.jpg)

**Comment by nick-keller** [comment, marked as answer]
This is not related to RDG but to React itself. You need to use a [useEffect](https://reactjs.org/docs/hooks-effect.html) hook:

```ts
  import { CellProps } from 'react-datasheet-grid'

  const TestEnter = (data: CellProps) => {
    useEffect(() => {
        if (data.focus && data.rowData.type === 0) {
          console.log('RUN function')
          data.stopEditing({ nextRow: false })
          setIsOpenDrawer(true)
        }
    }, [data.focus, data.rowData.type, data.stopEditing, setIsOpenDrawer])

    return <div>{data.rowData.test2}</div>
  }

```

**Reply by chairedevil** [reply]
Thank you so much for your response.
and sorry for my stupid question.

at first, I think I could use useEffect. but I didn't use it. 
I should try before I do ask.
